packages/modules/desktop-viewer-react/src/services/Initializer.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ClientRequestContext } from "@bentley/bentleyjs-core";
+import {
+  ElectronApp,
+  ElectronAppOpts,
+} from "@bentley/electron-manager/lib/ElectronFrontend";
+import { BentleyCloudRpcParams } from "@bentley/imodeljs-common";
+import { IModelApp } from "@bentley/imodeljs-frontend";
+import { UrlDiscoveryClient } from "@bentley/itwin-client";
+import { getIModelAppOptions, IModelBackendOptions } from "@itwin/viewer-react";
+
+import { DesktopViewerProps } from "../types";
+
+const getHostedConnectionInfo = async (
+  backendOptions?: IModelBackendOptions
+): Promise<BentleyCloudRpcParams> => {
+  const urlClient = new UrlDiscoveryClient();
+  const requestContext = new ClientRequestContext();
+
+  const orchestratorUrl = await urlClient.discoverUrl(
+    requestContext,
+    `iModelJsOrchestrator.K8S`,
+    backendOptions?.buddiRegion
+  );
+
+  if (backendOptions?.hostedBackend) {
+    if (!backendOptions.hostedBackend.title) {
+      //TODO localize
+      throw new Error("Please provide the title for the iTwin.js backend");
+    }
+    if (!backendOptions.hostedBackend.version) {
+      //TODO localize
+      throw new Error("Please provide a version for the iTwin.js backend");
+    }
+    return {
+      info: {
+        title: backendOptions.hostedBackend.title,
+        version: backendOptions.hostedBackend.version,
+      },
+      uriPrefix: orchestratorUrl,
+    };
+  } else {
+    return {
```

Comment:
i did make another comment about thsis in another file, but shouldnt desktop starter have a local backend that is used? and no need for gpb?